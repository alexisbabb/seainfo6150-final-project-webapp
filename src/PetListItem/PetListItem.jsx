import React, { useState } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import styles from "./PetListItem.module.css";
import PetImage from "../PetImage/PetImage.jsx";
import PetDetailsToggleButton from "../PetDetailsToggleButton/PetDetailsToggleButton.jsx";

const PetListItem = (props) => {
  const [isTextShowing, setIsTextShowing] = useState(false);

  function onClick() {
    setIsTextShowing(!isTextShowing);
  }

  return (
    <li className={styles.container}>
      <article className={styles.article}>
        <header className={styles.header}>
          <PetImage
            url={props.article.image.url}
            name={props.article.name}
          />
          <h2>
            <Link
              className={styles.link}
              to={`/petlist/${props.article.slug}`}
            >
              {props.article.name}
            </Link>
          </h2>
        </header>
        <div className={styles.wrapper}>
          {isTextShowing && (
            <div>
              <p>Breed: {props.article.breed}</p>
              <p>Age: {props.article.age}</p>
            </div>
          )}
        </div>
        <PetDetailsToggleButton
          buttonText={isTextShowing ? "Show less" : "Show more"}
          onClick={onClick}
        />
      </article>
    </li>
  );
};

PetListItem.propTypes = {
  article: PropTypes.object.isRequired,
};
export default PetListItem;