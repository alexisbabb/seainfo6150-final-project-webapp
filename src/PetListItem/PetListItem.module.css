.container { 
    border: 3px solid #b8610f;
    border-radius: 10%; 
    font-family: Arial, Helvetica, sans-serif;
    padding: 0.5em; 
    padding-bottom: 1em; 
  }
  
  .article {
      display: flex;
      flex-flow: column wrap;
      justify-content: space-between;
      height: 100%;
  }
  
  .header {
      display: flex;
      flex-flow: row wrap;
  }

  .link {
    color: #b8610f;
    font-family: 'Times New Roman', Times, serif; 
  }
  
  .wrapper {
      flex: 1;
      font-size: 1em; 
      margin: 0 0 1em;
  }

 
  


  


  