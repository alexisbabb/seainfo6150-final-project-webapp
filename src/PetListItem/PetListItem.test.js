import React from "react";
import PetListItem from "./PetListItem"; 

describe("PetListItem tests", () => {
    it("renders correctly", () => {
        const article = {
            "slug": "00001",
            "name": "Aladdin",
            "species": "Cat", 
            "breed": "Domestic Short Hair", 
            "gender": "Male", 
            "age": "2 Years", 
            "size": "Medium", 
            "color": "Black", 
            "location": "Newark", 
            "bio": "<p>Meet Aladdin! This sweet boy is looking for the perfect lap to relax in! He will follow you everywhere you go just like a dog. He is a very friendly boy that gets along with others cats. He originally came from a large cat colony living outside in lower Delaware. Aladdin is FIV positive, but that doesn't mean he won't live a long, normal life.</p>",
            "image": {
                "url": "https://pbs.twimg.com/media/DlmBBTXXcAEdn6V.jpg",
                "height": "400",
                "width": "400"
            }
        }; 
        const { container } = render(<PetListItem article={article} key = {article.slug}/>);
        expect(container).toMatchSnapshot();
    });
});