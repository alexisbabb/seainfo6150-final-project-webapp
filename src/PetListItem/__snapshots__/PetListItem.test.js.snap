// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetListItem tests renders correctly 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <header
        class="header"
      >
        <img
          alt="Aladdin"
          class="image"
          src="https://pbs.twimg.com/media/DlmBBTXXcAEdn6V.jpg"
        />
        <h2>
          <a
            class="link"
            href="/petlist/00001"
          >
            Aladdin
          </a>
        </h2>
      </header>
      <div
        class="wrapper"
      />
      <button
        class="detailsButton"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
